packages/ui/src/modal/acceptBid/AcceptBidModal.tsx

```diff
@@ -0,0 +1,376 @@
+import React, { ReactElement, useEffect, useState } from 'react'
+
+import {
+  Flex,
+  Box,
+  Text,
+  Anchor,
+  Button,
+  FormatEth,
+  FormatCurrency,
+  Loader,
+} from '../../primitives'
+
+// @ts-ignore
+import { Progress } from './Progress'
+import { Modal } from '../Modal'
+import {
+  faCircleExclamation,
+  faCheckCircle,
+} from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import TokenLineItem from '../TokenLineItem'
+import { AcceptBidStep, AcceptBidModalRenderer } from './AcceptBidModalRenderer'
+import Fees from './Fees'
+import { useTimeSince } from '../../hooks'
+
+type BidData = {
+  tokenId?: string
+  collectionId?: string
+  txHash?: string
+  maker?: string
+}
+
+type Props = Pick<Parameters<typeof Modal>['0'], 'trigger'> & {
+  tokenId?: string
+  collectionId?: string
+  onGoToToken?: () => any
+  onBidAccepted?: (data: BidData) => void
+  onAcceptanceError?: (error: Error, data: BidData) => void
+  onClose?: () => void
```

Comment:
`onClose` isn't used anywhere, we should call it when the modal is closed, refer to the other modals on how it's done.